control.lua

```diff
@@ -356,6 +357,8 @@ local function export_samples()
 			local settings = combinator.settings
 			if settings.name then
 				local entity_data = {
+                    id = entity.unit_number,
+                    surface = entity.surface.name,
```

Comment:
Surface names can change, and if you just want an id, `entity.surface.index` would be more robust:

```suggestion
                    unit_number = entity.unit_number,
                    surface_index = entity.surface.index,
```

Or did you need the name for your use-case?

Also, slight preference for sticking to the "unit_number" naming convention instead of "id".

Reply:
At least in the context of space exploration, the surface name is more visible and friendly. I don't have deep experience with the mod ecosystem, though, so I'm not sure how universal that is.

The name change is fine by me. It's just a name.